direct/core/src/test/java/cz/o2/proxima/direct/storage/InMemStorage.java

```diff
@@ -466,16 +465,23 @@ private void handleFlushDataBaseOnPosition(
           scheduler.scheduleAtFixedRate(
               onIdle, IDLE_FLUSH_TIME, IDLE_FLUSH_TIME, TimeUnit.MILLISECONDS);
       onIdleRef.set(onIdleFuture);
+      AtomicReference<StreamElement> lastConsumed = new AtomicReference<>();
       BiConsumer<StreamElement, OffsetCommitter> consumer =
           (el, committer) -> {
             try {
               if (!killSwitch.get()) {
                 synchronized (observer) {
                   el = cloneAndUpdateAttribute(getEntityDescriptor(), el);
                   watermark.update(el);
+                  Optional.ofNullable(lastConsumed.get())
```

Comment:
why do we need this - "mark as consumed on next element" approach?

Reply:
we need that to be sure that `observeBulkOffsets` will return the last consumed element (representing the offset).